### PR TITLE
CFE-2390 Document 'select_end_match_eof' body agent control option.

### PR DESCRIPTION
**Pull request by pasinskim**

### CODE REVIEW

**Comment by pasinskim** [comment]
@kacf here is missing documentation for https://github.com/cfengine/core/pull/2623.
@nickanderson @estenberg can you check the wording, please?

File: reference/promise-types/edit_line.markdown [hunk 1]
```diff
@@ -436,11 +436,13 @@ mark out the region of a file to be edited.
      {
      select_start => "\[$(x)\]";
      select_end => "\[.*\]";
-     select_end_match_eof => "false";
+     select_end_match_eof => "true";
      }
 ```
 
 If the `select_end` attribute is omitted, the selected region will run to the end of the file no matter what the value of `select_end_match_eof` is set to.
+Please note, that there is also global `body agent control` [`select_end_match_eof][cf-agent#select_end_match_eof] configuration option which is used to set the default behavior of
```
Comment (kacf) [inline reference/promise-types/edit_line.markdown]: ...also **a** global...

File: reference/promise-types/edit_line.markdown [hunk 2]
```diff
@@ -436,11 +436,13 @@ mark out the region of a file to be edited.
...
      select_start => "\[$(x)\]";
      select_end => "\[.*\]";
-     select_end_match_eof => "false";
+     select_end_match_eof => "true";
      }
 ```
 
 If the `select_end` attribute is omitted, the selected region will run to the end of the file no matter what the value of `select_end_match_eof` is set to.
+Please note, that there is also global `body agent control` [`select_end_match_eof][cf-agent#select_end_match_eof] configuration option which is used to set the default behavior of 
+`select_end_match_eof` for the entire policy. The local `edit_line` promise constraint is taking precedence over `body agent control` configuration option.
```
Comment (kacf) [inline reference/promise-types/edit_line.markdown]: The local `edit_line` promise constraint ~~is taking~~ **takes** precedence over **the** `body agent control` configuration option.

File: reference/components/cf-agent.markdown [hunk 3]
```diff
@@ -1067,6 +1067,26 @@ owned by a privileged user.
     }
 ```
 
+#### select_end_match_eof
+
+**Description:** When `select_end_match_eof` is set to true `select_end` will consider end of file as the end region if it is unable to match
+the end pattern. For more details see [`edit_line`][bundle edit_line] promise. 
+
+There is also [`select_end_match_eof`][bundle edit_line#select_end_match_eof] `edit_line` promise constraint which takes precedence over `body agent control` counterpart.
```
Comment (kacf) [inline reference/components/cf-agent.markdown]: ...over **the** `body agent control` counterpart.

**Comment by kacf** [comment]
Looks good to me, except for very minor grammatical stuff.

File: reference/components/cf-agent.markdown (hunk 3 above)
Comment (nickanderson) [inline reference/components/cf-agent.markdown]: I might change this to "**Note:** [`bundle edit_line select_end_match_eof`][bundle edit_line#select_end_match_eof] can override this setting at the individual promise level."

File: reference/promise-types/edit_line.markdown (hunk 1 above)
Comment (nickanderson) [inline reference/promise-types/edit_line.markdown]: I would suggest simplifying the sentence.

**Note:** The global `body agent control` [`select_end_match_eof][cf-agent#select_end_match_eof] sets the default behaviour for the entire policy.

**Comment by estenberg** [comment]
Yes, agreed. With the adjustments this is good to go!

**Comment by pasinskim** [comment]
All the issues fixed.

**Pull request merged**